src/TurbulenceClosures/turbulence_closure_implementations/TKEBasedVerticalDiffusivities/tke_dissipation_equations.jl

```diff
@@ -0,0 +1,305 @@
+using Oceananigans: fields
+using Oceananigans.Advection: div_Uc, U_dot_∇u, U_dot_∇v
+using Oceananigans.Fields: immersed_boundary_condition
+using Oceananigans.Grids: active_interior_map
+using Oceananigans.BoundaryConditions: apply_x_bcs!, apply_y_bcs!, apply_z_bcs!
+using Oceananigans.TimeSteppers: store_field_tendencies!, ab2_step_field!, implicit_step!
+using Oceananigans.TurbulenceClosures: ∇_dot_qᶜ, immersed_∇_dot_qᶜ, hydrostatic_turbulent_kinetic_energy_tendency
+using CUDA
+
+Base.@kwdef struct TKEDissipationEquations{FT}
+    Cᵋϵ :: FT = 1.92
+    Cᴾϵ :: FT = 1.44
+    Cᵇϵ :: FT = -0.65
+    Cᵂu★ :: FT = 1.0
+    CᵂwΔ :: FT = 1.0
+    Cᵂα  :: FT = 0.11
+    gravitational_acceleration  :: FT = 9.8065
```

Comment:
can we use the Oceananigans constant here?

Reply:
But wouldn't that mislead people into trying to change the constant in the code rather than specifically set this?

Reply:
I guess we still don't have a solid policy for constants and this has come up previously with no solution

Reply:
To clarify, say that someone wants to change `g` in their problem. If we take all constants from one place, peeople might think "oh, I can just fork the code and recompile Oceananigans with a different constant". 

But this is not what we want. We want people to specifically change the constant in their user code.

So then we also have to ask ourselves, why do we want to take constants from a central location?

If we want to do that, then we should probably use clima parameters instead.

Reply:
This isn't the first time it's come up so clearly we are missing something here

Reply:
ok, let's leave it as is for now ;)

Reply:
We may actually be able to excise this ... let me look into it a little more

Reply:
Maybe we can actually get rid of the Charnock length dependence here which I think is the best solution. Already there is the possibility of a mismatch between the surface flux model and this dissipation flux model which is not ideal